File: WinForms/VB/ProtectedContentEdit.vb
Imports DevExpress.Accessibility
Imports DevExpress.Utils
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace WindowsFormsApplication
    Public Class ProtectedContentEdit
        Inherits TextEdit
        Public Sub New()
            MyBase.New()
            Enabled = False
        End Sub
        Friend Const EditorName As String = "ProtectedContentEdit"
        Friend Const ProtectedContentText As String = "Protected Content"
        Shared Sub New()
            RepositoryItemProtectedContentTextEdit.Register()
        End Sub
        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return EditorName
            End Get
        End Property
    End Class

    Public Class RepositoryItemProtectedContentTextEdit
        Inherits RepositoryItemTextEdit

        Shared Sub New()
            Register()
        End Sub
        Public Sub New()
            ExportMode = ExportMode.DisplayText
        End Sub
        Friend Shared Sub Register()
            If Not EditorRegistrationInfo.[Default].Editors.Contains(ProtectedContentEdit.EditorName) Then
                EditorRegistrationInfo.[Default].Editors.Add(New EditorClassInfo(ProtectedContentEdit.EditorName, GetType(ProtectedContentEdit), GetType(RepositoryItemProtectedContentTextEdit), GetType(TextEditViewInfo), New TextEditPainter(), designTimeVisible:=True, EditImageIndexes.TextEdit, GetType(TextEditAccessible)))
            End If
        End Sub
        Public Overrides Function GetDisplayText(ByVal format As FormatInfo, ByVal editValue As Object) As String
            Return ProtectedContentEdit.ProtectedContentText
        End Function
        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return ProtectedContentEdit.EditorName
            End Get
        End Property
        Public Overrides Property [ReadOnly] As Boolean
            Get
                Return True
            End Get
            Set(ByVal value As Boolean)
            End Set
        End Property
    End Class
End Namespace

File: WinForms/CS/ProtectedContentEdit.cs
using DevExpress.ExpressApp.Editors;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Mask;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApplication {
	public class ProtectedContentEdit : StringEdit {
		static ProtectedContentEdit() {
			RepositoryItemProtectedContentTextEdit.Register();
		}
		public ProtectedContentEdit() { }
		public override string EditorTypeName { get { return RepositoryItemProtectedContentTextEdit.EditorName; } }
	}
	public class StringEdit : TextEdit {
		static StringEdit() {
			RepositoryItemStringEdit.Register();
		}
		public StringEdit() { }
		public StringEdit(int maxLength) {
			((RepositoryItemStringEdit)Properties).MaxLength = maxLength;
		}
		public override string EditorTypeName { get { return RepositoryItemStringEdit.EditorName; } }
	}
	public class RepositoryItemStringEdit : RepositoryItemTextEdit {
		internal const string EditorName = "StringEdit";
		internal static void Register() {
			if(!EditorRegistrationInfo.Default.Editors.Contains(EditorName)) {
				EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(StringEdit),
					typeof(RepositoryItemStringEdit),
					typeof(TextEditViewInfo), new TextEditPainter(), true, EditImageIndexes.TextEdit,
					typeof(DevExpress.Accessibility.TextEditAccessible)));
			}
		}
		static RepositoryItemStringEdit() {
			Register();
		}

		public override string EditorTypeName { get { return EditorName; } }
		public RepositoryItemStringEdit(int maxLength) : this() {
			MaxLength = maxLength;
		}
		public RepositoryItemStringEdit() {
			Mask.MaskType = MaskType.None;
			if(Mask.MaskType != MaskType.RegEx) {
				Mask.UseMaskAsDisplayFormat = true;
			}
		}
		public void Init(string displayFormat, string editMask, EditMaskType maskType) {
			Init(editMask, maskType);
			if(!string.IsNullOrEmpty(displayFormat)) {
				Mask.UseMaskAsDisplayFormat = false;
				DisplayFormat.FormatType = DevExpress.Utils.FormatType.Custom;
				DisplayFormat.FormatString = displayFormat;
			}
		}
		public void Init(string editMask, EditMaskType maskType) {
			if(!string.IsNullOrEmpty(editMask)) {
				Mask.EditMask = editMask;
				switch(maskType) {
					case EditMaskType.RegEx:
						Mask.UseMaskAsDisplayFormat = false;
						Mask.MaskType = MaskType.RegEx;
						break;
					default:
						Mask.MaskType = MaskType.Simple;
						break;
				}
			}
		}
	}
}
